### PR TITLE
[DDCE-5194] Play 3.0 upgrade

### PR DESCRIPTION
**Pull request by DannyPennington97**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -1,46 +1,29 @@
...
-    "uk.gov.hmrc"       %% "play-conditional-form-mapping" % "1.13.0-play-28",
-    "uk.gov.hmrc"       %% "domain"                        % "8.3.0-play-28",
-    "uk.gov.hmrc"       %% "bootstrap-frontend-play-28"    % bootstrapVersion
+    "uk.gov.hmrc"       %% "bootstrap-frontend-play-30"            % bootstrapVersion,
+    "uk.gov.hmrc"       %% "play-frontend-hmrc-play-30"            % "8.5.0",
+    "uk.gov.hmrc"       %% "play-conditional-form-mapping-play-30" % "2.0.0",
+    "uk.gov.hmrc"       %% "domain-play-30"                        % "9.0.0"
   )
 
   val test: Seq[ModuleID] = Seq(
```
Comment (machinesleet) [inline project/AppDependencies.scala:15]: From this test block I think you can remove scalatest, scalatestplus-play, scalacheck, and wiremock-standalone. Apart from wiremock-standalone they're all provided by bootstrap-test-play-30

File: build.sbt [hunk 2]
```diff
@@ -30,18 +22,12 @@ lazy val root = (project in file("."))
...
     ),
     PlayKeys.playDefaultPort := 8840,
-    ScoverageKeys.coverageExcludedFiles := "<empty>;Reverse.*;.*filters.*;.*handlers.*;.*components.*;.*repositories.*;" +
-      ".*BuildInfo.*;.*javascript.*;.*FrontendAuditConnector.*;.*Routes.*;.*GuiceInjector;" +
-      ".*ControllerConfiguration;.*LanguageSwitchController",
+    ScoverageKeys.coverageExcludedFiles := "<empty>;.*components.*;.*Mode.*;.*Routes.*;",
     ScoverageKeys.coverageMinimumStmtTotal := 91,
     ScoverageKeys.coverageFailOnMinimum := true,
     ScoverageKeys.coverageHighlighting := true,
     scalacOptions ++= Seq("-feature"),
```
Comment (frikit) [inline build.sbt:29]: replace line with 
```scala
scalacOptions ++= Seq(
      "-feature",
      "-Wconf:src=routes/.*:s",
      "-Wconf:cat=unused-imports&src=views/.*:s"
    ),
```

File: build.sbt [hunk 3]
```diff
@@ -60,19 +46,6 @@ lazy val root = (project in file("."))
...
-lazy val testSettings: Seq[Def.Setting[_]] = Seq(
-  fork        := true,
-  javaOptions ++= Seq(
-    "-Dconfig.resource=test.application.conf"
   )
-)
-
-dependencyOverrides ++= AppDependencies.overrides
 
 addCommandAlias("scalastyleAll", "all scalastyle test:scalastyle")
```
Comment (frikit) [inline build.sbt:51]: must be
```scala
addCommandAlias("scalastyleAll", "all scalastyle Test/scalastyle")
```

File: conf/logback.xml [hunk 4]
```diff
@@ -1,13 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <configuration>
```
Comment (frikit) [inline conf/logback.xml:2]: this file suppose to be https://github.com/hmrc/hmrc-frontend-scaffold.g8/blob/main/src/main/g8/conf/logback.xml and replace `new_service.log` with service name with underscores

File: project/AppDependencies.scala [hunk 5]
```diff
@@ -1,46 +1,25 @@
...
-    "uk.gov.hmrc"            %% "bootstrap-test-play-28"   % bootstrapVersion,
-    "org.scalatest"          %% "scalatest"                % "3.2.18",
-    "org.scalatestplus.play" %% "scalatestplus-play"       % "5.1.0",
+    "uk.gov.hmrc"            %% "bootstrap-test-play-30"   % bootstrapVersion,
     "org.scalatestplus"      %% "scalatestplus-scalacheck" % "3.1.0.0-RC2",
```
Comment (frikit) [inline project/AppDependencies.scala:17]: can be updated `"org.scalatestplus"           %% "scalacheck-1-17"          % "3.2.18.0",`

File: project/build.properties [hunk 6]
```diff
@@ -1 +1 @@
-sbt.version=1.8.2
+sbt.version=1.9.9
```
Comment (frikit) [inline project/build.properties:1]: as per platform this must be 1.9.7 to avoid issues

File: project/plugins.sbt [hunk 7]
```diff
@@ -2,17 +2,13 @@ resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefac
 resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 resolvers += Resolver.typesafeRepo("releases")
```
Comment (frikit) [inline project/plugins.sbt:3]: can be deleted

File: test/base/SpecBase.scala [hunk 8]
```diff
@@ -50,6 +50,12 @@ trait SpecBase extends PlaySpec
 
   lazy val fakeNavigator: FakeNavigator = new FakeNavigator()
 
+  val defaultAppConfigurations: Map[String, Any] = Map(
```
Comment (frikit) [inline test/base/SpecBase.scala:53]: this config must be used in all the places you built `new GuiceApplicationBuilder`

**Review by frikit — CHANGES_REQUESTED** [review]

File: project/plugins.sbt [hunk 9]
```diff
@@ -2,17 +2,13 @@ resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefac
...
+addSbtPlugin("uk.gov.hmrc"         %  "sbt-auto-build"       % "3.21.0")
+addSbtPlugin("uk.gov.hmrc"         %  "sbt-distributables"   % "2.5.0")
+addSbtPlugin("org.playframework"   %  "sbt-plugin"           % "3.0.3")
+addSbtPlugin("org.scalastyle"      % "scalastyle-sbt-plugin" % "1.0.0" exclude ("org.scala-lang.modules", "scala-xml_2.12"))
+addSbtPlugin("org.scoverage"       %  "sbt-scoverage"        % "2.0.11")
+addSbtPlugin("io.github.irundaia"  %  "sbt-sassify"          % "1.5.2")
+addSbtPlugin("net.ground5hark.sbt" %  "sbt-concat"           % "0.2.0")
+addSbtPlugin("com.typesafe.sbt"    %  "sbt-uglify"           % "2.0.0")
+addSbtPlugin("com.typesafe.sbt"    %  "sbt-digest"           % "1.1.4")
+addSbtPlugin("com.timushev.sbt"    %  "sbt-updates"          % "0.6.3")
```
Comment (frikit) [inline project/plugins.sbt:14]: can be updated

File: project/build.properties (hunk 6 above)
Comment (machinesleet) [reply project/build.properties:1]: there's a fair few completed migrations that have used 1.9.9
is there any issue here beyond longer build times?

Comment (frikit) [reply project/build.properties:1]: oh this works than fine , only issue might be only plugins incompatibility, but it sok leave it here then

**Review by frikit — APPROVED** [review]

**Pull request merged**